### PR TITLE
Refactor markups qt widgets

### PR DESCRIPTION
**Pull request by RafaelPalomar**
This PR addresses Slicer/Slicer#5857.

- [X] Renaming of `qSlicerMarkups*Widgets` to `qMRMLMarkups*Widgets`
- [X] Refactoring functionality in former `qSlicerMarkups*Widgets`
  - Create intermediate `qMRMLMarkupsAbstractOptionsWidgets` to keep the interface (pure virtual functions) used by `qSlicerMarkupsModuleWidget`
  - Make the new `qMRMLMarkups*Widget` independent components so they can be reused.
- [x] Fix internal update of `qMRMLMarkups*Widget` according to Slicer/Slicer#5857
- [x] Adapt Markups testing to the new qMRMLMarkups*Widgets
- [x] Rename `qMRMLMarkupsAdditional*OptionsWidget` classes to `qMRMLMarkups*OptionsWidget`

### CODE REVIEW

File: Modules/Loadable/Markups/Widgets/qMRMLMarkupsAdditionalOptionsWidgetsFactory.h [hunk 1]
```diff
@@ -23,65 +23,65 @@
...
   /// Registers an additional options widget.
   /// This factory object takes ownership of the widget (even if it fails to register it
   /// due to for example a widget already existing for that markup type), therefore the
   /// caller must not delete the widget.
-  Q_INVOKABLE bool registerAdditionalOptionsWidget(qSlicerMarkupsAdditionalOptionsWidget* widget);
+  Q_INVOKABLE bool registerAdditionalOptionsWidget(qMRMLMarkupsAdditionalOptionsWidget* widget);
 
   /// Unregisters an additional options widget.
-  Q_INVOKABLE bool unregisterAdditionalOptionsWidget(qSlicerMarkupsAdditionalOptionsWidget* widget);
+  Q_INVOKABLE bool unregisterAdditionalOptionsWidget(qMRMLMarkupsAdditionalOptionsWidget* widget);
```
Comment (lassoan) [inline Modules/Loadable/Markups/Widgets/qMRMLMarkupsAdditionalOptionsWidgetsFactory.h:57]: This appears to be not a "factory" just a storage of a few pre-created widgets. The factory should store function pointers or classes (preferable the latter) that can instantiate any number of the desired widgets.

**Review by lassoan — COMMENTED** [review]
Thank you, these changes are all useful. Reviewing the code uncovered some further needs for improving the design - see details in comments in the source code.

File: Modules/Loadable/Markups/Widgets/CMakeLists.txt [hunk 2]
```diff
@@ -19,17 +19,23 @@ set(${KIT}_SRCS
...
-  qSlicer${MODULE_NAME}ROIWidget.cxx
-  qSlicer${MODULE_NAME}ROIWidget.h
-  qSlicer${MODULE_NAME}AdditionalOptionsWidget_p.h
-  qSlicer${MODULE_NAME}AdditionalOptionsWidget.cxx
-  qSlicer${MODULE_NAME}AdditionalOptionsWidget.h
-  qSlicer${MODULE_NAME}AngleMeasurementsWidget.h
-  qSlicer${MODULE_NAME}AngleMeasurementsWidget.cxx
-  qSlicer${MODULE_NAME}CurveSettingsWidget.h
-  qSlicer${MODULE_NAME}CurveSettingsWidget.cxx
-  qSlicer${MODULE_NAME}AdditionalOptionsWidgetsFactory.h
```
[the remark this answers is not in the document]
Comment (RafaelPalomar) [reply Modules/Loadable/Markups/Widgets/CMakeLists.txt:31]: Addresed in 269bfbfaf9be88cfb353d344e50819fce2ee38c3

**Comment by lassoan** [comment]
Thanks a lot, I've reviewed and tested this, rebased on the latest Slicer master, and committed in https://github.com/Slicer/Slicer/commit/294e142189957e976caa76c9efe013d41f110d1c

**Pull request closed without merging**